main/src/chat/streaming.ts

```diff
@@ -21,210 +22,332 @@ export async function handleChatStreamRealtime(
   streamId: string,
   sender: Electron.WebContents
 ): Promise<void> {
-  try {
-    // Validate provider
-    const provider = CHAT_PROVIDERS.find((p) => p.id === request.provider)
-    if (!provider) {
-      throw new Error(`Unknown provider: ${request.provider}`)
-    }
+  return Sentry.startSpanManual(
+    {
+      name: 'Chat streaming request',
+      op: 'user.event',
+      attributes: {
+        'analytics.source': 'tracking',
+        'analytics.type': 'event',
+        stream_id: streamId,
+        provider: request.provider,
+        model: request.model,
+        start_timestamp: new Date().toISOString(),
+      },
+    },
+    async (span, finish) => {
+      try {
+        // Validate provider
+        const provider = CHAT_PROVIDERS.find((p) => p.id === request.provider)
+        if (!provider) {
+          throw new Error(`Unknown provider: ${request.provider}`)
+        }
+
+        // Create AI model
+        const model = provider.createModel(request.model, request.apiKey)
+        // Get MCP tools if enabled
+        const {
+          tools: mcpTools,
+          clients: mcpClients,
+          enabledTools,
+        } = await createMcpTools()
+
+        try {
+          const result = streamText({
+            model,
```

Comment:
nit: I would probably split the callback part in a separate function for readability